xrtl/testing/demo_main.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef XRTL_TESTING_DEMO_MAIN_H_
+#define XRTL_TESTING_DEMO_MAIN_H_
+
+namespace xrtl {
+namespace testing {
```

Comment:
I wanted to try to keep the number of top-level namespaces down a bit, and technically a demo is a kind of test (just not a unit test) so this seemed to fit. It also lets us share dependencies between tests and demos. Had the thought when I realized the demos we have should probably be marked testonly. For demos themselves my thought is that they can either be side-by-side in this repo (so window, window_test, window_demo) and/or in an external repo like we talked about, in which case they can live in whatever namespace they are logically in (ui for the former, some demos one for the latter).